media.go

```diff
@@ -313,6 +336,9 @@ func (pc *producerConsumer[T, U]) stop() {
 }
 
 func (pc *producerConsumer[T, U]) stopOne() {
+	_, span := trace.StartSpan(pc.cancelCtx, "gostream::producerConsumer::stopOne")
+	defer span.End()
```

Comment:
So I'm noticing `stop` calls called by this function aren't nested under this `stopOne` call. Is there any way to make that work? Could be helpful in debugging. 

I would look at the documentation for `context.WithCancel` to make sure the context created by this function call can replace `pc.cancelCtx` without screwing up preemption. If it does get screwed up you'll see noticeable hangs and leaked goroutines when you stop this stream or at a higher level, simply stop the Viam server.